app/models/decision.rb

```diff
@@ -1,50 +1,23 @@
 class Decision
   include ActiveModel::Model
 
-  attr_accessor :variant, :split, :admin
+  attr_accessor :split, :variant
 
-  validates :variant, presence: true, allow_blank: false
-  validates :split, presence: true
-  validates :admin, presence: true
-
-  validate :variant_belongs_to_split
+  validates :split, :variant, presence: true
 
   def save!
-    raise errors.full_messages.to_sentence unless valid?
-
-    weight_split_fully_to_variant
-    reassign_existing_assignments
-
-    true
-  end
-
-  def count
-    raise "count unavailable for unsaved Decision" unless bulk_assignment.persisted?
-    bulk_assignment.assignments.count
-  end
-
-  private
-
-  def weight_split_fully_to_variant
-    split.reweight!(variant => 100)
-  end
-
-  def reassign_existing_assignments
-    bulk_assignment.save!
-    existing_assignments.find_in_batches do |subset|
-      BulkReassignment.create!(assignments: subset, bulk_assignment: bulk_assignment)
+    raise "Variant must be present in the split" unless split.has_variant?(variant)
+    build_split_creation.save!.tap do
+      split.reload
     end
   end
 
-  def existing_assignments
-    @existing_assignments ||= Assignment.where(split: split).where.not(variant: variant)
-  end
-
-  def bulk_assignment
-    @bulk_assignment ||= BulkAssignment.new(admin: admin, split: split, reason: "Decision", variant: variant)
-  end
+  private
 
-  def variant_belongs_to_split
-    errors.add(:variant, "#{variant} is not a valid variant for split") unless split.has_variant? variant
+  def build_split_creation
+    split.build_split_creation(
```

Comment:
Normally the "creation" wording is fine if we're saying "i intend to create if it doesn't already exist, otherwise update it." But here what we're saying is "I know it already exists and I want to update it" -- I've started calling these classes `Upsert`s instead of `Creation`s, but I guess we gotta live with the current naming for this PR at least. 🤷

Reply:
I mean, we don't *have* to, but I am trying to keep the diff size down on this guy, but I like your thinking on that naming. I'll try to get that happening over time.

Reply:
so I pushed up a change to start hiding the underlying model naming by calling this `build_change` and making a `reconfigure!` method that does that plus self.reload. This is feeling on a good track. I'll rename the creation type to upsert in a subsequent PR.